astropy/utils/tests/test_data.py

```diff
@@ -1723,7 +1745,7 @@ def info(self):
         def read(self, length=None):
             return self.reader.read(length)
 
-    monkeypatch.setattr(urllib.request, "build_opener", mockurl_builder)
+    monkeypatch.setattr(astropy.utils.data, "_build_urlopener", mockurl_builder)
```

Comment:
I remember seeing monkeypatching of `astropy.utils.data` over at `astroquery`, so this is a heads up to `astroquery` people: @keflavich , @bsipocz , @ceb8